Makefile

```diff
@@ -120,3 +129,23 @@ bundle: manifests
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 	operator-sdk bundle validate ./bundle
+
+e2e_test: setup_system_test build
+	@find e2e-test -type f -name *_test.sh | xargs -I % sh -c "echo --- TEST % && bash %"
+
+setup_system_test: $(KIND_BIN)
+	@kubectl config use-context kind-espejo
+	@kubectl apply -k config/crd
+
+run_kind: setup_system_test
+	go run ./main.go
+
+$(KIND_BIN): $(TESTBIN_DIR)
+	curl -Lo $(KIND_BIN) "https://kind.sigs.k8s.io/dl/v$(KIND_VERSION)/kind-$$(uname)-amd64"
+	chmod +x $(KIND_BIN)
+	$(KIND_BIN) create cluster --name espejo
```

Comment:
Let's add `--kubeconfig` to the kind call and set a temporary `$KUBECONFIG`. I was quite surprised to find my local KUBECONFIG changed after running the tests...

Reply:
I have added some changes to the Makefile that should adjust it for all relevant targets